### PR TITLE
Update git instructions for contributing PRs through github.

### PR DESCRIPTION
**Pull request by spytheman**
Make instructions for contributing use `https://github.com/YOUR_GITHUB_USERNAME/v` , 
which is friendlier to beginners that do not have ssh keys setup.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -58,31 +58,31 @@ maps), `time/`, `os/`, etc. Their documentation is pretty clear.
 username will be referred to later as 'YOUR_GITHUB_USERNAME'. Change it
 accordingly in the steps below.)
 
-1. Clone https://github.com/vlang/v in a folder, say nv (`git clone
-https://github.com/vlang/v nv`)
+1. Fork https://github.com/vlang/v using github's interface in your own account, lets say that the forked repository is at
```
Comment (medvednikov) [inline CONTRIBUTING.md]: GitHub

**Comment by medvednikov** [comment]
Should also add hub or git cli instructions to avoid using the slow web interface.

**Comment by spytheman** [comment]
I agree. The hub cli is awesome!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -58,31 +58,78 @@ maps), `time/`, `os/`, etc. Their documentation is pretty clear.
...
-1. When finished with a feature/bugfix, you can: `git checkout -b fix_alabala`
-1. `git push pullrequest`  # (NOTE: the pullrequest remote was setup on step 3)
-1. On GitHub's web interface, I go to: https://github.com/vlang/v/pulls  Here
-the UI shows a nice dialog with a button to make a new pull request based on
-the new pushed branch. (Example dialogue:
-https://url4e.com/gyazo/images/364edc04.png)
-1. After making your pullrequest (aka, PR), you can continue to work on the
-branch... just do step #5 when you have more commits.
-1. If there are merge conflicts, or a branch lags too much behind V's master,
+1. Fork https://github.com/vlang/v using GitHub's interface in your own account, lets say that the forked repository is at
```
Comment (danieldaeschle) [inline CONTRIBUTING.md]: "to your own account"?

Comment (spytheman) [reply CONTRIBUTING.md]: Thanks. Changed.

Comment (danieldaeschle) [reply CONTRIBUTING.md]: "Let's say..."

And it should also be in a new sentence.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -58,31 +58,78 @@ maps), `time/`, `os/`, etc. Their documentation is pretty clear.
...
+`https://github.com/YOUR_GITHUB_USERNAME/v` .
+2. Clone the main v repository https://github.com/vlang/v in a local folder on your computer, say named nv/
```
Comment (danieldaeschle) [inline CONTRIBUTING.md]: "to a local folder"

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -58,31 +58,78 @@ maps), `time/`, `os/`, etc. Their documentation is pretty clear.
...
+4. `git remote add pullrequest https://github.com/YOUR_GITHUB_USERNAME/v`  
+NB: the remote named `pullrequest` should point to YOUR own forked repo, not the main v repository!
+After this, your local cloned repository is prepared for making pullrequests, and you can just do normal git operations such as: `git pull` `git status` and so on.
+
+1. When finished with a feature/bugfix/change, you can: 
+`git checkout -b fix_alabala`
+2. `git push pullrequest`  # (NOTE: the `pullrequest` remote was setup on step 4)
+3. On GitHub's web interface, go to: https://github.com/vlang/v/pulls  
+
+Here the UI shows a nice dialog with a button to make a new pull request based on
```
Comment (danieldaeschle) [inline CONTRIBUTING.md]: remove "nice"

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -58,31 +58,78 @@ maps), `time/`, `os/`, etc. Their documentation is pretty clear.
...
+the new pushed branch. 
+(Example dialogue: https://url4e.com/gyazo/images/364edc04.png)
```
Comment (danieldaeschle) [inline CONTRIBUTING.md]: not "Dialog"? Hmm, maybe difference in BE and AE.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -58,31 +58,78 @@ maps), `time/`, `os/`, etc. Their documentation is pretty clear.
...
-main V repository's master, then `git checkout master; git pull --rebase origin
-master` work as expected (this is actually used by `v up`) and it can always do
-so cleanly. Git is very flexible, so there may be simpler/easier ways to
-accomplish the same thing.
+   1. `git pull --rebase origin master` # solve conflicts and do `git rebase --continue`
+   2. `git push pullrequest -f` # this will overwrite your current remote branch with the updated version of your changes.
+
+The point of doing the above steps, is to never directly push to the main V
+repository, *only to your own fork*. Since your local `master` branch tracks the
+main V repository's master, then `git checkout master`, as well as `git pull --rebase origin master` will continue to work as expected (these are actually used by `v up`) and git can always do so cleanly.
```
Comment (danieldaeschle) [inline CONTRIBUTING.md]: that "so" at the end sounds strange...

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -58,31 +58,78 @@ maps), `time/`, `os/`, etc. Their documentation is pretty clear.
...
+
+Git is very flexible, so there are other ways to accomplish the same thing.
+
+## Using Github's hub CLI tool
+
+You can download the `hub` tool from https://hub.github.com/ . Using
+`hub`, you will not need to go through the (sometimes) slow website
```
Comment (danieldaeschle) [inline CONTRIBUTING.md]: you won't need

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -58,31 +58,78 @@ maps), `time/`, `os/`, etc. Their documentation is pretty clear.
...
+7. `hub pull-request`
+
+Optionally, you can track the status of your PR CI tests with:
+
+8. `hub ci-status --verbose`
+
+### Fixing failing tests:
+If everything is OK, after 5-10 minutes, the CI tests should pass for
+all platforms. If not, visit the URLs for the failing CI jobs, see
+which tests have failed, then fix them, by making more changes. Just use
```
Comment (danieldaeschle) [inline CONTRIBUTING.md]: have failed and then fix them by making more changes

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -58,31 +58,78 @@ maps), `time/`, `os/`, etc. Their documentation is pretty clear.
...
+
+Optionally, you can track the status of your PR CI tests with:
+
+8. `hub ci-status --verbose`
+
+### Fixing failing tests:
+If everything is OK, after 5-10 minutes, the CI tests should pass for
+all platforms. If not, visit the URLs for the failing CI jobs, see
+which tests have failed, then fix them, by making more changes. Just use 
+`git push pullrequest` to publish your changes, and the CI tests will
```
Comment (danieldaeschle) [inline CONTRIBUTING.md]: remove that ","

File: CONTRIBUTING.md (hunk 3 above)
Comment (spytheman) [reply CONTRIBUTING.md]: done

**Pull request merged**